plugins/com.google.cloud.tools.eclipse.appengine.validation/src/com/google/cloud/tools/eclipse/appengine/validation/MissingRuntime.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.eclipse.appengine.validation;
+
+/**
+ * No runtime element in appengine-web.xml.
+ */
+class MissingRuntime extends ElementProblem {
+
+  MissingRuntime() {
+    super("A runtime element is required.",
+        "com.google.cloud.tools.eclipse.appengine.validation.runtimeMarker");
```

Comment:
This runtimeMarker needs to be defined in the plugin.xml

Reply:
done